### DISCUSSION TITLE
potential regression between 4.4.1 and 4.12.0

### DISCUSSION THREAD

**Discussion by freddyrios** [dotnet/dotNext#164] [General]
While running some tests with the latest version (4.12) we experienced the cluster had more often leader changes. Additionally, it seems that competing leader decisions being applied to a MemoryBasedStateMachine during a leader change happen more often.

Are there some changes that could relate to the above?

The test was done without other changes (besides removing the metrics configuration due to the breaking change), only comparing 4.12 vs. 4.41.

**Comment by sakno** [comment]
I see no radical changes:
* `ProtocolStream` was reused for UDP transport as well (which is not your case)
* Deprecation of metrics
* Automatic removal of faulty cluster members using custom or predefined (Phi Accrual) failure detector. However, this is disabled by default
* Reduced memory allocations
* Fixed bug when `ForceReplicationAsync` can be completed earlier than the expected log entry replicates

The last fix is critical if you have the following code:
```csharp
// add uncommitted log entry A
await cluster.ForceReplicationAsync();
// expect that A is committed (replicated)
```

**Reply by freddyrios** [reply]
It might then be the timing differences that change the situation in the specific scenario being tested. Will let you know if I find more from our side.
